modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/TransactionStateResolver.java

```diff
@@ -201,14 +204,15 @@ private void resolveDistributiveTxState(
         } else {
             assert localMeta.txState() == ABANDONED : "Unexpected transaction state [txId=" + txId + ", txStateMeta=" + localMeta + ']';
```

Comment:
This is impossible because we have ABANDONED state in memory. If we have this state, that means we have locks, which we cannot face with write intents.

Reply:
Why do you think it's impossible? As long as we have this state, we should take into account the possibility of meeting it.